### PR TITLE
Feature/sl 5085 inject meta in xlsx

### PR DESCRIPTION
**Pull request by gcione**

### CODE REVIEW

**Review by andlambert — CHANGES_REQUESTED** [review]

File: tests/test_inject_meta_in_xlsx.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from lib.data_structures.status import Status
+from lib.data_structures.xls_column import XlsColumn
+from lib.process import inject_meta_in_xlsx
+
+
+def test_inject_meta_in_xlsx():
+    dest_folder = join('.', '')
+    row = 0
+
+    expected = []
```
[the remark this answers is not in the document]
Comment (gcione) [reply tests/test_inject_meta_in_xlsx.py:17]: Done

File: lib/process/inject_meta_in_xlsx.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+
+def create_metadata_file(dest_folder: str, metadata: List[MetadataModel]):
+    workbook = xlsxwriter.Workbook(join(dest_folder, 'metadata.xlsx'))
+    worksheet = workbook.add_worksheet()
+
+    row = 1
+
+    # Headers
+    worksheet.write(XlsColumn.DOC_TYPE.value + str(row), "doc_type")
```
[the remark this answers is not in the document]
Comment (gcione) [reply lib/process/inject_meta_in_xlsx.py:17]: Done

File: lib/process/inject_meta_in_xlsx.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    worksheet.write(XlsColumn.STATUS.value + str(row), "status")
+    worksheet.write(XlsColumn.DATE_ARR.value + str(row), "date_arr")
+    worksheet.write(XlsColumn.PART.value + str(row), "part")
+    worksheet.write(XlsColumn.NUM_ARR.value + str(row), "num_arr")
+    worksheet.write(XlsColumn.NUM_ROL.value + str(row), "num_rol")
+    worksheet.write(XlsColumn.DICTUM.value + str(row), "dictum")
+
+    for i in range(0, len(metadata)):
+        line = metadata[i]
+        if isinstance(line, MetadataModel):
```
[the remark this answers is not in the document]
Comment (gcione) [reply lib/process/inject_meta_in_xlsx.py:29]: Done

File: lib/process/inject_meta_in_xlsx.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    worksheet.write(XlsColumn.LANG.value + str(row), "lang")
+    worksheet.write(XlsColumn.FILENAME.value + str(row), "filename")
+    worksheet.write(XlsColumn.STATUS.value + str(row), "status")
+    worksheet.write(XlsColumn.DATE_ARR.value + str(row), "date_arr")
+    worksheet.write(XlsColumn.PART.value + str(row), "part")
+    worksheet.write(XlsColumn.NUM_ARR.value + str(row), "num_arr")
+    worksheet.write(XlsColumn.NUM_ROL.value + str(row), "num_rol")
+    worksheet.write(XlsColumn.DICTUM.value + str(row), "dictum")
+
+    for i in range(0, len(metadata)):
```
[the remark this answers is not in the document]
Comment (gcione) [reply lib/process/inject_meta_in_xlsx.py:27]: Done

File: tests/test_inject_meta_in_xlsx.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    expected.append(MetadataModel(DocType.ORDONNANCE.value, Lang.NL.value, "14604Dep.pdf", Status.EXT_OK.value))
+    expected.append(MetadataModel(DocType.ORDONNANCE.value, Lang.FR.value, "14407Dep.pdf", Status.EXT_OK.value))
+    expected.append(MetadataModel(DocType.ARRET.value, Lang.DE.value, "251410d.pdf", Status.EXT_OK.value))
+    inject_meta_in_xlsx.create_metadata_file(dest_folder, expected)
+
+    xlsx_reader = xlrd.open_workbook(join(dest_folder, 'metadata.xlsx'))
+    xlsx_sheet = xlsx_reader.sheet_by_index(0)
+
+    # Test headers
+    assert xlsx_sheet.cell_value(row, XlsColumn.INDEX_DOC_TYPE.value) == "doc_type"
```
[the remark this answers is not in the document]
Comment (gcione) [reply tests/test_inject_meta_in_xlsx.py:29]: Done

File: lib/data_structures/xls_column.py [hunk 6]
```diff
@@ -11,3 +11,13 @@ class XlsColumns(Enum):
     NUM_ARR = 'G'
     NUM_ROL = 'H'
     DICTUM = 'I'
+
+    INDEX_DOC_TYPE = 0
```
[the remark this answers is not in the document]
Comment (gcione) [reply lib/data_structures/xls_column.py:15]: Done

**Review by andlambert — CHANGES_REQUESTED** [review]

File: lib/data_structures/xls_column_index.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
+from enum import IntEnum
+
+
+class XlsColumnIndex(IntEnum):
+    INDEX_DOC_TYPE = 0
```
[the remark this answers is not in the document]
Comment (gcione) [reply lib/data_structures/xls_column_index.py]: Done

**Review by andlambert — APPROVED** [review]

**Pull request merged**